### PR TITLE
[feate] Added support for postgresql based on dev-1.4.0

### PR DESCRIPTION
**Pull request by sjgllgh**
### What is the purpose of the change
- Added postgresql support for linkis metadata.

### Related issues/PRs
- Related issues: #4190 

### Brief change log
- Modify MybatisConfigurationFactory.java for support multiple data sources.
- Modify linkis-cg-linkismanager.properties, linkis-mg-gateway.properties and linkis-cg-linkismanager.properties  for support multiple data sources.
- To better distinguish between different data sources, change the original mapper file path. For example, move resources/mapper/common/contextHistoryMapper.xml to resources/mapper/common/mysql/contextHistoryMapper.xml
- Add mapper files for postgresql. For example, resources/mapper/common/postgresql/contextHistoryMapper.xml
- Some test classes were added or modified. 

### How to use postgresql
1. pom.xml
- remove <scope>test<scpoe> in postgresql's dependency.
2. linkis.properties
- Add database connection informations.
- Add linkis.server.mybatis.pagehelper.dialect=postgresql. The configuration detault value is mysql.
3. linkis-cg-linkismanager.properties, linkis-mg-gateway.properties and linkis-cg-linkismanager.properties
- Select mapper files that postgresql can execute.
4. If you want to use postgresql in test classes, You should change the properties used by the test class.
- Add database connection informations.
- Select mapper files that postgresql can execute.
- Note: Please use the existing postgresql database to test.

### Checklist
- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ x] I have explained the need for this PR and the problem it solves
- [ x] I have explained the changes or the new features added to this PR
- [ x] I have added tests corresponding to this change
- [ x] I have updated the documentation to reflect this change
- [ x] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ x] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+spring.datasource.password=
+spring.datasource.schema=classpath:create.sql
+
+#pgtest
+#spring.datasource.driver-class-name=org.postgresql.Driver
+#spring.datasource.url=jdbc:postgresql://${host}:${port}/linkis
+#spring.datasource.username=linkis
+#spring.datasource.password=linkis
+
+mybatis-plus.mapper-locations=classpath*:mapper/common/mysql/*.xml
```
Comment (jackxu2011) [inline linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties:32]: mapper/common, is for general,  the mysql should move to path  mapper/mysql,  and pg to mapper/postgresql.
almost all database support sql92.  so unless the sql used a database specific function,  others can be shared.

the mapper-locations can have more the one,  classpath*:/mapper/common/*.xml,  classpath*:/mapper/mysql/*.xml,

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/ExternalResourceProviderDaoImpl.xml
Comment (jackxu2011) [inline linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/ExternalResourceProviderDaoImpl.xml:1]: the same problem

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+spring.datasource.url=jdbc:h2:mem:test;MODE=MySQL;DB_CLOSE_DELAY=-1;DATABASE_TO_LOWER=true
+spring.datasource.username=sa
+spring.datasource.password=
+spring.datasource.schema=classpath:create.sql
+
+#pgtest
+#spring.datasource.driver-class-name=org.postgresql.Driver
+#spring.datasource.url=jdbc:postgresql://${host}:${port}/linkis
+#spring.datasource.username=linkis
+#spring.datasource.password=linkis
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties]: Thank you for your suggestion. I have already used h2's pg compatibility mode in test. However, there are some tests that are not compatible with both mysql and psql. I have commented on these tests.

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties (hunk 1 above)
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/application.properties]: Thank you for your suggestion. I have modified it according to your suggestion.

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/linkis.properties [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+#
+wds.linkis.test.mode=true
+wds.linkis.server.mybatis.datasource.url=
+wds.linkis.server.mybatis.datasource.username=
+wds.linkis.server.mybatis.datasource.password=
+wds.linkis.log.clear=true
+wds.linkis.server.version=v1
+##mybatis
+wds.linkis.server.mybatis.mapperLocations=classpath*:org/apache/linkis/manager/dao/impl/*.xml
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/linkis.properties:24]: Thank you for your suggestion. I have modified it according to your suggestion.

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/EngineConnBmlResourceMapper.xml
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/EngineConnBmlResourceMapper.xml:1]: Thank you for your suggestion. I have modified it according to your suggestion.

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/ExternalResourceProviderDaoImpl.xml
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/mysql/ExternalResourceProviderDaoImpl.xml:1]: Thank you for your suggestion. I have modified it according to your suggestion.

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/postgresql/EngineConnBmlResourceMapper.xml
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/postgresql/EngineConnBmlResourceMapper.xml:1]: Thank you for your suggestion. I have modified it according to your suggestion.

**Comment by aiceflower** [comment]
Please resolve the conflict

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/linkis.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (jackxu2011) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/test/resources/linkis.properties:24]: the config should be modified

File: linkis-public-enhancements/linkis-datasource/linkis-metadata/src/test/resources/create_pg.sql [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+DROP TABLE IF EXISTS "linkis_ps_datasource_lineage";
+CREATE TABLE linkis_ps_datasource_lineage (
+	id int4 NOT NULL,
+	table_id int8 NULL,
+	source_table varchar(64) NULL,
+	update_time timestamp(6) NULL,
+	CONSTRAINT linkis_mdq_lineage_pkey PRIMARY KEY (id)
+);
+
+CREATE SEQUENCE linkis_ps_datasource_table_id_seq INCREMENT BY 1 MINVALUE 1 MAXVALUE 2147483647 START 1 CACHE 1 NO CYCLE;
```
Comment (jackxu2011) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata/src/test/resources/create_pg.sql:76]: also this place,  use the pg's auto increment type?

File: linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            <if test="startId != null">and id >= #{startId}</if>
+            <if test="instances != null">and instances = #{instances}</if>
+        </where>
+        ORDER BY linkis_ps_job_history_group_history.created_time DESC
+    </select>
+
+    <!-- // todo check -->
+    <update id="updateJobHistory" flushCache="true" parameterType="org.apache.linkis.jobhistory.entity.JobHistory">
+        UPDATE linkis_ps_job_history_group_history
+        <trim prefix="set" suffixOverrides=",">
```
Comment (jackxu2011) [inline linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml:147]: use `<set>` replace

File: pom.xml [hunk 6]
```diff
@@ -1361,7 +1362,17 @@
       <artifactId>spring-test</artifactId>
       <scope>test</scope>
     </dependency>
-
+    <dependency>
+      <groupId>org.yaml</groupId>
+      <artifactId>snakeyaml</artifactId>
+      <version>${snakeyaml.version}</version>
+    </dependency>
```
Comment (jackxu2011) [inline pom.xml:1369]: remove the snakeyaml dependency

File: pom.xml [hunk 7]
```diff
@@ -238,6 +238,7 @@
     <spotless-maven-plugin.version>2.24.1</spotless-maven-plugin.version>
     <jacoco-maven-plugin.version>0.8.8</jacoco-maven-plugin.version>
     <jacoco.skip>false</jacoco.skip>
+    <postgresql.connector.version>42.2.20</postgresql.connector.version>
```
Comment (jackxu2011) [inline pom.xml:241]: move the property to db section

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceInfoMapper.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        <![CDATA[INSERT INTO "linkis_ps_instance_info"("instance", "name")
+        VALUES(#{instance}, #{applicationName}) on conflict(instance) do update set name=excluded.name]]>
+    </insert>
+
+    <delete id="removeInstance" parameterType="org.apache.linkis.common.ServiceInstance">
+        DELETE FROM linkis_ps_instance_info WHERE instance = #{instance}
+    </delete>
+
+    <update id="updateInstance" parameterType="org.apache.linkis.instance.label.entity.InstanceInfo">
+        UPDATE linkis_ps_instance_info SET
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceInfoMapper.xml]: done, thank your advice

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceLabelMapper.xml [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    </sql>
+
+    <sql id="label_insert_columns">
+        "label_key", "label_value", "label_feature",
+        "label_value_size"
+    </sql>
+    <select id="selectForUpdate" parameterType="Integer" resultMap="insPersistenceLabelMap">
+        <![CDATA[SELECT ]]>
+        <include refid="label_search_columns" />
+        <![CDATA[ FROM "linkis_ps_instance_label" WHERE "id" = #{labelId} FOR UPDATE;]]>
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceLabelMapper.xml]: done, thank your advice

File: linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml (hunk 5 above)
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml]: done, thank your advice

File: pom.xml (hunk 7 above)
Comment (sjgllgh) [reply pom.xml]: done, thank your advice

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml [hunk 10]
```diff
@@ -46,10 +46,10 @@
         FROM linkis_cg_engine_conn_plugin_bml_resources
         <where>
             <if test="engineConnType != null and engineConnType != ''">
-                <![CDATA[AND `engine_conn_type` like concat('%', #{engineConnType}, '%')]]>
+                <![CDATA[AND engine_conn_type like concat('%', #{engineConnType}, '%')]]>
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml]: done, thanks for your advice

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml [hunk 11]
```diff
@@ -46,10 +46,10 @@
...
             </if>
             <if test="engineConnVersion != null and engineConnVersion != ''">
-                <![CDATA[AND `version` like concat('%', #{engineConnVersion}, '%')]]>
+                <![CDATA[AND version like concat('%', #{engineConnVersion}, '%')]]>
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml]: done, thanks for your advice

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml [hunk 12]
```diff
@@ -46,10 +46,10 @@
...
             </if>
         </where>
         <![CDATA[ ORDER BY create_time DESC ]]>
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml]: done, thanks for your advice

File: linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml [hunk 13]
```diff
@@ -58,25 +58,25 @@
...
         DELETE FROM linkis_cg_engine_conn_plugin_bml_resources
         WHERE engine_conn_type = #{engineConnBmlResource.engineConnType}
-        AND `version` = #{engineConnBmlResource.version}
+        AND version = #{engineConnBmlResource.version}
         AND file_name = #{engineConnBmlResource.fileName}
     </delete>
 
     <update id="update" parameterType="org.apache.linkis.engineplugin.server.entity.EngineConnBmlResource">
         UPDATE linkis_cg_engine_conn_plugin_bml_resources
         <trim prefix="set" suffixOverrides=",">
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/resources/mapper/common/EngineConnBmlResourceMapper.xml]: done, thanks for your advice

File: linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/ECResourceRecordMapper.xml [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+        INSERT INTO linkis_cg_ec_resource_info_record(label_value,create_user, ticket_id, service_instance, request_times, request_resource,
+                                                        used_times, used_resource, release_times, released_resource,release_time,used_time,ecm_instance,log_dir_suffix,status,metrics)
+        VALUES (#{labelValue},#{createUser},#{ticketId},#{serviceInstance},#{requestTimes},
+        #{requestResource},#{usedTimes},#{usedResource},
+        #{releaseTimes},#{releasedResource},#{releaseTime},#{usedTime},#{ecmInstance},#{logDirSuffix},#{status},#{metrics})
+    </insert>
+
+    <update id="updateECResourceInfoRecord" flushCache="true" parameterType="org.apache.linkis.manager.common.entity.persistence.ECResourceInfoRecord">
+        UPDATE linkis_cg_ec_resource_info_record
+        <trim prefix="set" suffixOverrides=",">
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/ECResourceRecordMapper.xml]: done, thanks for your advice

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceInfoMapper.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceInfoMapper.xml]: done, thanks for your advice

File: pom.xml (hunk 7 above)
Comment (sjgllgh) [reply pom.xml]: done, thanks for your advice

File: pom.xml (hunk 6 above)
Comment (sjgllgh) [reply pom.xml]: done, thanks for your advice

File: linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml (hunk 5 above)
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-jobhistory/src/main/resources/mapper/postgresql/JobHistoryMapper.xml]: done, thanks for your advice

File: linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceLabelMapper.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-instance-label/linkis-instance-label-server/src/main/resources/mapper/postgresql/InstanceLabelMapper.xml]: done, thanks for your advice

File: linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/ECResourceRecordMapper.xml [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+        </where>
+        ORDER BY linkis_cg_ec_resource_info_record.create_time DESC
+    </select>
+
+    <select id="getECResourceInfoList" resultType="org.apache.linkis.manager.common.entity.persistence.ECResourceInfoRecord" >
+        select * from linkis_cg_ec_resource_info_record
+        where id in (
+        <!-- choose the latest record of this ec -->
+        select max(ecr.id) as id from linkis_cg_ec_resource_info_record ecr
+        WHERE 1=1
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/ECResourceRecordMapper.xml]: done

File: linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/LabelManagerMapper.xml [hunk 16]
```diff
@@ -0,0 +1,549 @@
...
+        ) t
+        INNER JOIN linkis_cg_manager_service_instance si ON si.instance = t.service_instance
+    </select>
+
+    <!--<select id="getNodeByLabelKeyValue1" resultType="org.apache.linkis.common.ServiceInstance">
+        SELECT si.instance ,si.name as "application_name" FROM linkis_cg_manager_label l ,
+        linkis_cg_manager_label_service_instance lsi ,linkis_cg_manager_service_instance si
+        WHERE l.id = lsi.label_id AND si.instance = lsi.service_instance and
+        l.label_key = #{labelKey} and l.label_value = #{stringValue}
+    </select>-->
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/main/resources/mapper/postgresql/LabelManagerMapper.xml]: done

File: linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/test/java/org/apache/linkis/manager/dao/LabelManagerMapperTest.java [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+    labelManagerMapper.batchDeleteResourceByLabelId(notBlankIds);
+  }
+
+  //    //    没有实现
+  //    @Test
+  //    public void batchDeleteResourceByLabelIdInDirect(){
+  //        List<Integer> notBlankIds = new ArrayList<>();
+  //        notBlankIds.add(1);
+  //        labelManagerMapper.batchDeleteResourceByLabelIdInDirect(notBlankIds);
+  //    }
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-computation-governance/linkis-manager/linkis-manager-persistence/src/test/java/org/apache/linkis/manager/dao/LabelManagerMapperTest.java]: done

File: linkis-dist/package/conf/linkis-mg-gateway.properties [hunk 18]
```diff
@@ -21,6 +21,7 @@ wds.linkis.gateway.conf.url.pass.auth=/dss/
 wds.linkis.gateway.conf.enable.token.auth=true
 wds.linkis.is.gateway=true
 wds.linkis.server.mybatis.mapperLocations=classpath*:mapper/common/*.xml,classpath*:mapper/mysql/*.xml
+#wds.linkis.server.mybatis.mapperLocations=classpath*:mapper/common/*.xml,classpath*:mapper/postgresql/*.xml
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-dist/package/conf/linkis-mg-gateway.properties]: done, this conf in linkis-cg-linkismanager.properties, linkis-mg-gateway.properties and linkis-ps-publicservice.properties has been moved to conf/linkis.properties

File: linkis-engineconn-plugins/shell/pom.xml
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-engineconn-plugins/shell/pom.xml:1]: nothing has changed in this file

File: linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/ConfigurationConfigKeyMapper.xml [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+        </selectKey>
+        INSERT INTO linkis_ps_configuration_config_key (
+        key, description, name, engine_conn_type, default_value, validate_type, validate_range, is_hidden, is_advanced, level, "treeName")
+        VALUES (
+        #{key}, #{description}, #{name}, #{engineConnType}, #{defaultValue}, #{validateType}, #{validateRange}, cast(#{hidden} as bool), cast(#{advanced} as bool), #{level}, #{treeName})
+    </insert>
+
+    <update id="updateById" parameterType="org.apache.linkis.basedatamanager.server.domain.ConfigurationConfigKey">
+        UPDATE linkis_ps_configuration_config_key
+        <trim prefix="set" suffixOverrides=",">
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/ConfigurationConfigKeyMapper.xml]: done

File: linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/DatasourceTypeKeyMapper.xml [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+        description,description_en,value_regex,
+        ref_id,ref_value,data_source,
+        update_time,create_time
+    </sql>
+
+    <select id="getListByPage" resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"></include>
+        <![CDATA[ from linkis_ps_dm_datasource_type_key ]]>
+        where 1 = 1
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/DatasourceTypeKeyMapper.xml:56]: done

File: linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/DatasourceTypeMapper.xml [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+    <sql id="Base_Column_List">
+        "id","name","description",
+        "option","classifier","icon",
+        "layers"
+    </sql>
+
+    <select id="getListByPage" resultMap="BaseResultMap">
+        SELECT
+        <include refid="Base_Column_List"></include>
+        <![CDATA[ from linkis_ps_dm_datasource_type ]]>
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-basedata-manager/src/main/resources/mapper/postgresql/DatasourceTypeMapper.xml]: done

File: linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/postgresql/UserTenantMapper.xml [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+    </select>
+
+    <delete id="deleteTenant">
+        DELETE FROM linkis_cg_tenant_label_config
+        WHERE id = cast(#{id} as int)
+    </delete>
+
+    <update id="updateTenant" parameterType="org.apache.linkis.configuration.entity.TenantVo">
+        UPDATE linkis_cg_tenant_label_config
+        <trim prefix="set" suffixOverrides=",">
```
[the remark this answers is not in the document]
Comment (sjgllgh) [reply linkis-public-enhancements/linkis-configuration/src/main/resources/mapper/postgresql/UserTenantMapper.xml]: done

**Review by jackxu2011 — APPROVED** [review]
LGTM

**Comment by jackxu2011** [comment]
ping @peacewong

**Pull request merged**